### PR TITLE
PROD-408: Notification commands and script

### PR DESCRIPTION
**Pull request by neel-ds**
The PR adds a feature to subscribe notifications and a script to drop a general message to all subscribers.

The mock shows that the format of message is not preserved from CLI. The sender needs to manually add expressions to reserve the format of message (e.g line breaks)
Screenshot 2024-10-30 at 2 36 25 AM
Screenshot 2024-10-30 at 2 36 53 AM

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,45 +19,19 @@ const WEB_APP_URL = process.env.WEB_APP_URL || '';
 
 const bot = new Telegraf(BOT_TOKEN);
 
-const miniAppUrl = WEB_APP_URL;
+const miniAppUrl = WEB_APP_URL + '/bot';
```
Comment (marvinmarnold) [inline src/index.ts]: Remove

File: src/index.ts [hunk 2]
```diff
@@ -11,50 +16,25 @@ import * as nodeCrypto from 'crypto';
 const BOT_TOKEN = process.env.BOT_TOKEN || '';
 const ENVIRONMENT = process.env.NODE_ENV || '';
 const WEB_APP_URL = process.env.WEB_APP_URL || '';
+const LOGIN_PATH = process.env.LOGIN_PATH || '';
```
Comment (marvinmarnold) [inline src/index.ts:19]: This can probably be a constant, since it won't ever depend on environment.

File: src/index.ts [hunk 3]
```diff
@@ -64,55 +44,71 @@ const getMiniAppButton = (ctx: any) => {
...
+
+  // Get all subscribed users
+  const subscribedUsers = await getSubscribedUsers();
+
+  // Auto-subscribe new users
+  if (
+    !subscribedUsers.find((user) => user.telegramUsername === username) &&
+    username
+  ) {
+    await updateUserNotification(ctx, true);
```
Comment (marvinmarnold) [inline src/index.ts:71]: Why don't we set the notification status in `createUserByTelegram`?

**Review by marvinmarnold — COMMENTED** [review]
Screenshot 2024-11-12 at 1 16 13 PM

This isn't working

File: src/index.ts (hunk 3 above)
Comment (neel-ds) [reply src/index.ts:71]: Yes! It sets notifications by default for new users. Removed it

**Pull request merged**